Content.Shared/Weapons/Reflect/SharedReflectSystem.cs

```diff
@@ -160,4 +166,55 @@ private void OnReflectHitscan(EntityUid uid, ReflectComponent component, ref Hit
 
         return true;
     }
+
+    private void OnReflectEquipped(EntityUid uid, ReflectComponent comp, GotEquippedEvent args)
+    {
+
+        if (!TryComp(args.Equipee, out ReflectComponent? reflection))
+            return;
+
+        reflection.Enabled = true;
+
+        // reflection probability should be: (1 - old probability) * newly-equipped item probability + old probability
+        // example: if entity has .25 reflection and newly-equipped item has .7, entity should have (1 - .25) * .7 + .25 = .775
+        reflection.ReflectProb += (1 - reflection.ReflectProb) * comp.ReflectProb;
+
+    }
+
+    private void OnReflectUnequipped(EntityUid uid, ReflectComponent comp, GotUnequippedEvent args)
+    {
+
+        if (!TryComp(args.Equipee, out ReflectComponent? reflection))
+            return;
+
+        if (comp.ReflectProb == 1)
+        {
+            float newProb = 1;
+
+            foreach (var reflector in GetSlots(args.Equipee))
+            {
+                if (!TryComp(reflector, out ReflectComponent? refcomp))
+                    continue;
+
+                var prob = refcomp.ReflectProb;
+                newProb -= newProb * prob;
+            }
+            reflection.ReflectProb = 1 - newProb;
+        }
+        else
+        {
+            // to undo the above example, take .775 minus .7, then divide by 1 minus .7:
+            // (.775 - .7) / (1 - .7) = .075/.3 = .25 (+ float error)
+            try
+            {
+                reflection.ReflectProb = (reflection.ReflectProb - comp.ReflectProb) / (1 - comp.ReflectProb);
+            }
+            catch (DivideByZeroException e)
+            {
+                reflection.ReflectProb = 0;
+                Logger.Error("component with reflectprob 1 should've been pruned", e.ToString());
+            }
+        }
```

Comment:
Trycatch is for unexpected error handling and is also too slow for games, just handle it yourself.